tests/test_penalties.py

```diff
@@ -759,3 +763,44 @@ def test_penalty(self, random_ragged_cmf):
 
         with pytest.raises(TypeError):
             penalty.penalty(A)
+
+
+class TestTemporalSmoothness(BaseTestFactorMatricesPenalty):
+    PenaltyType = penalties.TemporalSmoothnessPenalty
+    penalty_default_kwargs = {"smoothness_l": 1}
+
+    n_rows = 10
+    min_rows = n_rows
+    max_rows = n_rows
+
+    def test_penalty(self, random_regular_cmf):
+        # Check that the penalty term is computed correctly.
+
+        penalty = self.PenaltyType(**self.penalty_default_kwargs)
+        cmf, random_ragged_shapes, rank = random_regular_cmf
+        weights, (A, B_is, C) = cmf
+
+        K = len(B_is)
+
+        penalty_term = 0
+        for k in range(1, K):
+            penalty_term += tl.norm(B_is[k] - B_is[k - 1]) ** 2
+
+        assert np.isclose(penalty.penalty(B_is), penalty_term)
+
+    def get_invariant_matrices(self, rng, shapes):
+        # Generate a list of invariant matrices that will not be changed by the proximal operator.
+
+        K = len(shapes)
+
+        invariant_matrix = np.array(rng.random_sample(shapes[0]))
+        invariant_matrices = [np.copy(invariant_matrix) for _ in range(K)]
```

Comment:
We need to wrap this in `tl.tensor` to be sure that it works for PyTorch too

```suggestion
        invariant_matrices = [tl.tensor(invariant_matrix.copy()) for _ in range(K)]
```